clang/test/dpct/helper_files_ref/include/blas_utils.hpp

```diff
@@ -385,6 +385,56 @@ inline void rk_impl(cl::sycl::queue &q, oneapi::mkl::uplo uplo,
   }
 }
 
+template <class Ta, class Tb, class Ts>
+inline void
+trsm_batch_impl(cl::sycl::queue &q, oneapi::mkl::side left_right,
+                oneapi::mkl::uplo upper_lower, oneapi::mkl::transpose trans,
+                oneapi::mkl::diag unit_diag, int m, int n, const void *alpha,
+                const void **a, int lda, void **b, int ldb, int batch_size) {
+  struct mat_info_t {
+    mat_info_t(oneapi::mkl::side side_info, oneapi::mkl::uplo uplo_info,
+               oneapi::mkl::transpose transpose_info,
+               oneapi::mkl::diag diag_info, Ts value_info, std::int64_t m,
+               std::int64_t n, std::int64_t lda, std::int64_t ldb,
+               std::int64_t groupsize_info)
+        : side_info(side_info), uplo_info(uplo_info),
+          transpose_info(transpose_info), diag_info(diag_info),
+          value_info(value_info), groupsize_info(groupsize_info) {
+      size_info[0] = m;
+      size_info[1] = n;
+      ld_info[0] = lda;
+      ld_info[1] = ldb;
+    }
+    oneapi::mkl::side side_info;
+    oneapi::mkl::uplo uplo_info;
+    oneapi::mkl::transpose transpose_info;
+    oneapi::mkl::diag diag_info;
+    Ts value_info;
+    std::int64_t size_info[2];
+    std::int64_t ld_info[2];
+    std::int64_t groupsize_info;
+  };
+
+  Ts alpha_value = dpct::get_value(reinterpret_cast<const Ts *>(alpha), q);
+
+  mat_info_t *mat_info =
+      new mat_info_t(left_right, upper_lower, trans, unit_diag, alpha_value, m,
+                     n, lda, ldb, batch_size);
+
+  cl::sycl::event e = oneapi::mkl::blas::column_major::trsm_batch(
+      q, &(mat_info->side_info), &(mat_info->uplo_info),
+      &(mat_info->transpose_info), &(mat_info->diag_info), mat_info->size_info,
+      mat_info->size_info + 1, &(mat_info->value_info),
+      reinterpret_cast<const Ta **>(a), mat_info->ld_info,
+      reinterpret_cast<Tb **>(b), mat_info->ld_info + 1, 1,
+      &(mat_info->groupsize_info));
+
+  q.submit([&](cl::sycl::handler &cgh) {
+    cgh.depends_on(e);
+    cgh.host_task([=] { delete mat_info; });
+  });
+}
+
 }
```

Comment:
I guess, this is the ending of detail namespace. If so, please add a comment "// detail" just to improve readability a bit.

Reply:
OK. I will do that.